### DISCUSSION TITLE
argocd shows OutOfSync (requires pruning) when we remove featureGates: -UseStrimziPodSets and stop using stateful sets

### DISCUSSION THREAD

**Discussion by dbaarorapr** [strimzi/strimzi-kafka-operator#9111] [General]
argocd shows OutOfSync (requires pruning) when we remove featureGates: -UseStrimziPodSets and stop using stateful sets

so all we have done is to stop using the stateful sets  and start using the StrimziPodSets and our argocd application which was perfectly fine and now has resources which are OutOfSync (requires pruning)  messages

I have seen many threads and  even setting the deleteclaim: true does not make it go away for PVC and many ControllerRevision which are also  OutOfSync (requires pruning)   

Strimzi Version : 0.33.2
Kafka Version:3.4.0
ArgoCD  Version:

{
    "Version": "v2.4.7+81630e6",
    "BuildDate": "2022-07-18T21:17:35Z",
    "GitCommit": "81630e6d5075ac53ac60457b51343c2a09a666f4",
    "GitTreeState": "clean",
    "GoVersion": "go1.18.4",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KustomizeVersion": "v4.4.1 2021-11-11T23:36:27Z",
    "HelmVersion": "v3.8.1+g5cb9af4",
    "KubectlVersion": "v0.23.1",
    "JsonnetVersion": "v0.18.0"
}

**Comment by dbaarorapr** [comment]
@scholzj  ^^^ we have an issue above and our CRB is showing out of sync with argocd
we can get around by using the deleteClaim: True with Kafka CR and PVC;s shows up as synced as owner references show up in the PVC
Also all is good for us ,  when we use statefulsets with strimzi 0.33.2 but they break when we remove the feature gate and forced to use SPS

We need to upgrade to 0.36.1 strimzi but SPS cannot be turned off now and if we keep it ON ,then argocd shows out of sync for CRB and PVC

**Reply by scholzj** [reply]
I'm not Argo user and I know nothing about it, sorry. I think in the past the advice was for example to use something called annotation-based tracking or something like that. Another was I think to add some annotations to the PVC to instruct Argo to ignore them. But that is likely covered in many of the existing threads here, on Slack, in issues etc.
